lib/psm.cc

```diff
@@ -724,6 +741,9 @@ static rpmRC dbAdd(rpmts ts, rpmte te)
     headerPutUint32(h, RPMTAG_INSTALLTIME, &installTime, 1);
     headerPutUint32(h, RPMTAG_INSTALLCOLOR, &tscolor, 1);
 
+    if (auxh)
+	mergeAux(auxh, h);
```

Comment:
What is `auxh` is NULL (possible due to `rpmteHeaderAux()` potentially returning NULL if `verifyPackage()` never ran and thus never initialized `te->auxh`)? It seems like this could crash in `headerSort()` in such a case.

Reply:
rpmteHeaderAux() is called with init==1 in dbAdd() so it will always return non-NULL, but that doesn't make much sense and is actually anoter refactoring artifact (I initially thought to move all the installtime etc to the aux header too, in which case the init==1 would've been necessary, but then decided to keep this minimal here). I'll fix that.

But, mergeAux() can't get called with a NULL auxh because the code is testing for non-NULL auxh right there? :sweat_smile:

Reply:
Dropped the superfluous auxh init from dbAdd()